Support for Arm64

**Describe the bug**
Using ChromeOS with Crostini which is a debian type linux. No binaries available for this architecture so I tried to build from source.

Bottom line after running `make install`, the last few lines output were: 
```
Architecture aarch64 is not supported by this installation script
Makefile:22: recipe for target 'install-tools' failed
make: *** [install-tools] Error 1
```
**To Reproduce**
Steps to reproduce the behavior:
1. Launch terminal on ChromeOS/Crostini
2. Run following for dependencies:
```
sudo apt-get install libgl1-mesa-dev
sudo apt-get install xorg-dev
```
3. Then run `go get -u github.com/liamg/aminal`
4. Then `cd $GOPATH/github.com/liamg/aminal`
5. Then `make install`
6. Output is:
```
go test -v ./...
?   	github.com/liamg/aminal	[no test files]
=== RUN   TestOffsets
--- PASS: TestOffsets (0.00s)
=== RUN   TestBufferCreation
--- PASS: TestBufferCreation (0.00s)
=== RUN   TestBufferWriteIncrementsCursorCorrectly
--- PASS: TestBufferWriteIncrementsCursorCorrectly (0.00s)
=== RUN   TestWritingNewLineAsFirstRuneOnWrappedLine
--- PASS: TestWritingNewLineAsFirstRuneOnWrappedLine (0.00s)
=== RUN   TestWritingNewLineAsSecondRuneOnWrappedLine
--- PASS: TestWritingNewLineAsSecondRuneOnWrappedLine (0.00s)
=== RUN   TestSetPosition
--- PASS: TestSetPosition (0.00s)
=== RUN   TestMovePosition
--- PASS: TestMovePosition (0.00s)
=== RUN   TestVisibleLines
--- PASS: TestVisibleLines (0.00s)
=== RUN   TestClearWithoutFullView
--- PASS: TestClearWithoutFullView (0.00s)
=== RUN   TestClearWithFullView
--- PASS: TestClearWithFullView (0.00s)
=== RUN   TestCarriageReturn
--- PASS: TestCarriageReturn (0.00s)
=== RUN   TestCarriageReturnOnFullLine
--- PASS: TestCarriageReturnOnFullLine (0.00s)
=== RUN   TestCarriageReturnOnFullLastLine
--- PASS: TestCarriageReturnOnFullLastLine (0.00s)
=== RUN   TestCarriageReturnOnWrappedLine
--- PASS: TestCarriageReturnOnWrappedLine (0.00s)
=== RUN   TestCarriageReturnOnLineThatDoesntExist
--- PASS: TestCarriageReturnOnLineThatDoesntExist (0.00s)
=== RUN   TestGetCell
--- PASS: TestGetCell (0.00s)
=== RUN   TestGetCellWithHistory
--- PASS: TestGetCellWithHistory (0.00s)
=== RUN   TestGetCellWithBadCursor
--- PASS: TestGetCellWithBadCursor (0.00s)
=== RUN   TestCursorAttr
--- PASS: TestCursorAttr (0.00s)
=== RUN   TestCursorPositionQuerying
--- PASS: TestCursorPositionQuerying (0.00s)
=== RUN   TestRawPositionQuerying
--- PASS: TestRawPositionQuerying (0.00s)
=== RUN   TestEraseLine
--- PASS: TestEraseLine (0.00s)
=== RUN   TestEraseLineToCursor
--- PASS: TestEraseLineToCursor (0.00s)
=== RUN   TestEraseLineAfterCursor
--- PASS: TestEraseLineAfterCursor (0.00s)
=== RUN   TestEraseDisplay
--- PASS: TestEraseDisplay (0.00s)
=== RUN   TestEraseDisplayToCursor
--- PASS: TestEraseDisplayToCursor (0.00s)
=== RUN   TestEraseDisplayFromCursor
--- PASS: TestEraseDisplayFromCursor (0.00s)
=== RUN   TestBackspace
--- PASS: TestBackspace (0.00s)
=== RUN   TestHorizontalResizeView
--- PASS: TestHorizontalResizeView (0.00s)
=== RUN   TestLine
--- PASS: TestLine (0.00s)
PASS
ok  	github.com/liamg/aminal/buffer	(cached)
=== RUN   TestColourTomlEncoding
--- PASS: TestColourTomlEncoding (0.00s)
=== RUN   TestColourTomlUnmarshalling
--- PASS: TestColourTomlUnmarshalling (0.00s)
=== RUN   TestKeyCombinations
--- PASS: TestKeyCombinations (0.00s)
PASS
ok  	github.com/liamg/aminal/config	(cached)
?   	github.com/liamg/aminal/glfont	[no test files]
?   	github.com/liamg/aminal/gui	[no test files]
?   	github.com/liamg/aminal/hints	[no test files]
=== RUN   TestParsing
--- PASS: TestParsing (0.00s)
PASS
ok  	github.com/liamg/aminal/sixel	(cached)
?   	github.com/liamg/aminal/terminal	[no test files]
?   	github.com/liamg/aminal/version	[no test files]
go vet -v
internal/race
internal/cpu
errors
math/bits
runtime/internal/sys
unicode/utf8
encoding
unicode
sync/atomic
image/color
internal/testlog
unicode/utf16
runtime/internal/atomic
container/list
crypto/internal/subtle
crypto/subtle
internal/bytealg
vendor/golang_org/x/crypto/cryptobyte/asn1
vendor/golang_org/x/net/dns/dnsmessage
internal/nettrace
math
strconv
crypto/rc4
runtime/cgo
runtime
sync
io
internal/singleflight
reflect
hash
bytes
math/rand
syscall
hash/crc32
sort
strings
bufio
encoding/binary
internal/syscall/unix
crypto/cipher
time
image
path
encoding/base64
regexp/syntax
github.com/liamg/aminal/vendor/github.com/gobuffalo/syncx
crypto/aes
internal/poll
image/internal/imageutil
image/draw
crypto
crypto/des
crypto/internal/randutil
crypto/sha512
crypto/hmac
crypto/md5
crypto/sha1
crypto/sha256
encoding/pem
vendor/golang_org/x/crypto/poly1305
regexp
vendor/golang_org/x/crypto/internal/chacha20
vendor/golang_org/x/crypto/curve25519
vendor/golang_org/x/text/transform
github.com/liamg/aminal/vendor/go.uber.org/zap/buffer
github.com/liamg/aminal/vendor/go.uber.org/atomic
os
vendor/golang_org/x/crypto/chacha20poly1305
github.com/liamg/aminal/vendor/go.uber.org/zap/internal/bufferpool
fmt
github.com/liamg/aminal/vendor/github.com/mcuadros/go-version
path/filepath
flag
context
encoding/hex
log
github.com/liamg/aminal/vendor/go.uber.org/zap/internal/exit
os/exec
encoding/json
compress/flate
io/ioutil
compress/gzip
github.com/liamg/aminal/vendor/github.com/kr/pty
github.com/liamg/aminal/vendor/github.com/joho/godotenv
github.com/liamg/aminal/vendor/github.com/BurntSushi/toml
github.com/liamg/aminal/vendor/github.com/go-gl/gl/all-core/gl
github.com/liamg/aminal/vendor/github.com/pkg/errors
github.com/liamg/aminal/vendor/github.com/go-gl/glfw/v3.2/glfw
github.com/liamg/aminal/vendor/github.com/gobuffalo/envy
net/url
math/big
vendor/golang_org/x/text/unicode/bidi
crypto/rand
encoding/asn1
net
crypto/rsa
crypto/elliptic
crypto/dsa
crypto/x509/pkix
crypto/ecdsa
vendor/golang_org/x/crypto/cryptobyte
vendor/golang_org/x/text/secure/bidirule
vendor/golang_org/x/text/unicode/norm
net/textproto
vendor/golang_org/x/net/http2/hpack
crypto/x509
vendor/golang_org/x/net/idna
mime
vendor/golang_org/x/net/http/httpguts
mime/quotedprintable
vendor/golang_org/x/net/http/httpproxy
net/http/internal
github.com/liamg/aminal/vendor/github.com/markbates/oncer
crypto/tls
mime/multipart
github.com/liamg/aminal/hints
net/http/httptrace
github.com/liamg/aminal/config
github.com/liamg/aminal/vendor/golang.org/x/image/math/fixed
github.com/liamg/aminal/sixel
github.com/liamg/aminal/vendor/go.uber.org/multierr
github.com/liamg/aminal/vendor/golang.org/x/image/font
github.com/liamg/aminal/vendor/github.com/golang/freetype/raster
github.com/liamg/aminal/vendor/go.uber.org/zap/internal/color
github.com/liamg/aminal/buffer
os/user
github.com/liamg/aminal/vendor/go.uber.org/zap/zapcore
net/http
github.com/liamg/aminal/vendor/github.com/golang/freetype/truetype
github.com/liamg/aminal/vendor/github.com/gobuffalo/packd
github.com/liamg/aminal/version
github.com/liamg/aminal/vendor/go.uber.org/zap
github.com/liamg/aminal/vendor/github.com/golang/freetype
github.com/liamg/aminal/vendor/github.com/gobuffalo/packr
github.com/liamg/aminal/glfont
github.com/liamg/aminal/vendor/github.com/riywo/loginshell
github.com/liamg/aminal/terminal
github.com/liamg/aminal/gui
github.com/liamg/aminal
# github.com/liamg/aminal
Checking file ./config.go
Checking file ./logger.go
Checking file ./main.go
which dep || curl -L https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  5110  100  5110    0     0  22892      0 --:--:-- --:--:-- --:--:-- 22914
Architecture aarch64 is not supported by this installation script
Makefile:22: recipe for target 'install-tools' failed
make: *** [install-tools] Error 1
```

**Expected behavior**
Clean install for aminal!

**Environment (please complete the following information):**
```
$ cat /etc/*release
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

--- a/gui/gui.go
+++ b/gui/gui.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"os/exec"
 	"runtime"
+	"strconv"
 	"strings"
 	"time"
 
@@ -110,7 +111,7 @@ func (gui *GUI) Render() error {
 
 	gui.logger.Debugf("Creating window...")
 	var err error
-	gui.window, err = gui.createWindow(gui.width, gui.height)
+	gui.window, err = gui.createWindow()
 	if err != nil {
 		return fmt.Errorf("Failed to create window: %s", err)
 	}
@@ -148,6 +149,7 @@ func (gui *GUI) Render() error {
 			gui.terminal.SetDirty()
 		}
 	})
+
 	w, h := gui.window.GetFramebufferSize()
 	gui.resize(gui.window, w, h)
 
@@ -322,27 +324,77 @@ Buffer Size: %d lines
 
 }
 
-func (gui *GUI) createWindow(width int, height int) (*glfw.Window, error) {
+func (gui *GUI) createWindow() (*glfw.Window, error) {
 	if err := glfw.Init(); err != nil {
 		return nil, fmt.Errorf("Failed to initialise GLFW: %s", err)
 	}
 
 	glfw.WindowHint(glfw.Resizable, glfw.True)
-	glfw.WindowHint(glfw.ContextVersionMajor, 4)
-	glfw.WindowHint(glfw.ContextVersionMinor, 1)
 	glfw.WindowHint(glfw.OpenGLProfile, glfw.OpenGLCoreProfile)
 	glfw.WindowHint(glfw.OpenGLForwardCompatible, glfw.True)
 
-	window, err := glfw.CreateWindow(width, height, "Terminal", nil, nil)
-	if err != nil {
-		return nil, fmt.Errorf("Failed to create window: %s", err)
+	versions := [][2]int{
+		{4, 6},
+		{4, 5},
+		{4, 4},
+		{4, 3},
+		{4, 2},
+		{4, 1},
+		{4, 0},
+		{3, 3},
+		{3, 2},
 	}
+
+	var window *glfw.Window
+
+	for _, v := range versions {
+		var err error
+		window, err = gui.createWindowWithOpenGLVersion(v[0], v[1])
+		if err != nil {
+			gui.logger.Warnf("Failed to create window: %s. Will attempt older version...", err)
+		}else{
+			break
+		}
+	}
+
+	if window == nil {
+		return nil, fmt.Errorf("failed to create window, please update your graphics drivers and try again")
+	}
+
 	window.SetSizeLimits(300, 150, 10000, 10000)
 	window.MakeContextCurrent()
+	window.Show()
+	window.Focus()
 
 	return window, nil
 }
 
+func(gui *GUI) createWindowWithOpenGLVersion(major int, minor int) (*glfw.Window, error) {
+
+	glfw.WindowHint(glfw.ContextVersionMajor, major)
+	glfw.WindowHint(glfw.ContextVersionMinor, minor)
+
+	window, err := glfw.CreateWindow(gui.width, gui.height, "Terminal", nil, nil)
+	if err != nil {
+		e := err.Error()
+		if i := strings.Index(e, ", got version "); i > - 1 {
+			v := strings.Split(strings.TrimSpace(e[i+14:]), ".")
+			if len(v) == 2 {
+				major, err := strconv.Atoi(v[0])
+				if err == nil {
+					if minor, err := strconv.Atoi(v[1]); err == nil {
+						return gui.createWindowWithOpenGLVersion(major, minor)
+					}
+
+				}
+			}
+		}
+
+		return nil, fmt.Errorf("Failed to create window using OpenGL v%d.%d: %s.", major, minor, err)
+	}
+
+	return window, nil
+}
 // initOpenGL initializes OpenGL and returns an intiialized program.
 func (gui *GUI) createProgram() (uint32, error) {
 	if err := gl.Init(); err != nil {

